milla/src/lib.rs

```diff
@@ -0,0 +1,2224 @@
+const ATMOS_SLEEPING_AGENT: usize = 10;
+const ATMOS_AGENT_B: usize = 11;
+// Thermal energy, in joules.
+const ATMOS_THERMAL_ENERGY: usize = 12;
+// The thermal conductivity in each direction. Starts at OPEN_HEAT_TRANSFER_COEFFICIENT, but can be
+// reduced by the tile and its contents.
+const ATMOS_SUPERCONDUCTIVITY_EAST: usize = 13;
+const ATMOS_SUPERCONDUCTIVITY_NORTH: usize = 14;
+const ATMOS_SUPERCONDUCTIVITY_WEST: usize = 15;
+const ATMOS_SUPERCONDUCTIVITY_SOUTH: usize = 16;
+// How much heat capacity the tile itself adds, in joules per kelvin, in addition to any air on it.
+const ATMOS_INNATE_HEAT_CAPACITY: usize = 17;
+
+// How many gases are there?
+const GAS_COUNT: usize = ATMOS_AGENT_B - ATMOS_OXYGEN + 1;
+// Where is the first one?
+const GAS_OFFSET: usize = ATMOS_OXYGEN;
+// The total number of atmos fields.
+const ATMOS_DEPTH: usize = ATMOS_INNATE_HEAT_CAPACITY + 1;
+
+// The two axes, Y and X. The order is arbitrary, but will break the copy from active to inactive
+// if changed.
+const AXES: [(i32, i32); 2] = [(0, 1), (1, 0)];
+// The four cardinal directions. The order is arbitrary, and doesn't matter.
+const DIRECTIONS: [(i32, i32); 4] = [(0, 1), (1, 0), (0, -1), (-1, 0)];
+
+// The specific heat of each gas, in joules per kelvin-mole.
+// The numbers here are completely wrong for actual gases, but they're what LINDA used, so we'll
+// keep them for now.
+const SPECIFIC_HEAT_OXYGEN: f32 = 20.0;
+const SPECIFIC_HEAT_CARBON_DIOXIDE: f32 = 30.0;
+const SPECIFIC_HEAT_NITROGEN: f32 = 20.0;
+const SPECIFIC_HEAT_TOXINS: f32 = 200.0;
+const SPECIFIC_HEAT_SLEEPING_AGENT: f32 = 40.0;
+const SPECIFIC_HEAT_AGENT_B: f32 = 300.0;
+
+// Convenience array, so we can add loop through gases and calculate heat capacity.
+const SPECIFIC_HEATS: [f32; GAS_COUNT] = [
+    SPECIFIC_HEAT_OXYGEN,
+    SPECIFIC_HEAT_CARBON_DIOXIDE,
+    SPECIFIC_HEAT_NITROGEN,
+    SPECIFIC_HEAT_TOXINS,
+    SPECIFIC_HEAT_SLEEPING_AGENT,
+    SPECIFIC_HEAT_AGENT_B,
+];
+
+// Plasmafire constants, see usage for details.
+const PLASMA_BURN_MIN_TEMP: f32 = 100.0 + T0C;
+const PLASMA_BURN_OPTIMAL_TEMP: f32 = 1370.0 + T0C;
+const PLASMA_BURN_MAX_OXYGEN_AVAILABILITY: f32 = 10.0;
+const PLASMA_BURN_MAX_RATIO: f32 = 0.25;
+const PLASMA_BURN_WORST_OXYGEN_PER_PLASMA: f32 = 1.4;
+const PLASMA_BURN_BEST_OXYGEN_PER_PLASMA: f32 = 0.4;
+
+// How much thermal energy each reaction produces, in joules per mole of "fuel", whichever gas that
+// is.
+const AGENT_B_CONVERSION_ENERGY: f32 = 20_000.0;
+const NITROUS_BREAKDOWN_ENERGY: f32 = 200_000.0;
+const PLASMA_BURN_ENERGY: f32 = 3_000_000.0;
+
+// The flag that's used to determine which buffer is active.
+//
+// Data type explanation:
+// * OnceLock is how we make a safe static variable, by ensuring it will be initialized exactly
+//   once.
+// * Mutex lets us read and write that variable safely across threads.
+// * It's fundamentally a boolean flag.
+static BUFFER_FLIPPER: OnceLock<Mutex<bool>> = OnceLock::new();
```

Comment:
Is there a reason for using a bool here over a quick enum? Might be a little clearer, especially if you were to match on it

Reply:
Done.